src/main/java/liqp/Template.java

```diff
@@ -275,68 +249,194 @@ public ParseTree getParseTree() {
 
     /**
      * Returns a new Template instance from a given input string.
+     * 
+     * Important: This method may have undesired side-effects from globally defined insertions and
+     * filters. Please use {@link TemplateParser} instead.
      *
      * @param input
-     *         the input string holding the Liquid source.
+     *            the input string holding the Liquid source.
      *
      * @return a new Template instance from a given input string.
+     * @deprecated use {@link TemplateParser#parse(String)}
      */
+    @Deprecated
```

Comment:
This method is handy, easy, simple, expected, and even in the library samples were a lot of its usage. A static method with global scope is ok. Library provides a choice. Either custom instance with custom tags, filters. etc, which you can construct as you want. Either change the global one on your risk.  
Somehow disagree with moving out of default global scope and static methods.
@bkiers attention required.